common/src/main/java/tachyon/util/network/NetworkAddressUtils.java

```diff
@@ -119,15 +204,27 @@ public static String getLocalIpAddress(int timeout) {
       // Make sure that the address is actually reachable since in some network configurations
       // it is possible for the InetAddress.getLocalHost() call to return a non-reachable
       // address e.g. a broadcast address
-      if (address.isLoopbackAddress() || !address.isReachable(timeout)) {
+      if (address.isAnyLocalAddress() || address.isLoopbackAddress()
+          || !address.isReachable(timeout) || !(address instanceof Inet4Address)) {
         Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+
+        // Make getNetworkInterfaces have the same order of network interfaces as listed on
+        // unix-like systems. This optimization can help avoid to get some special addresses, such
+        // as loopback address"127.0.0.1", virtual bridge address "192.168.122.1" as far as
+        // possible.
+        if (!sIsWindows) {
+          List<NetworkInterface> netIFs = Collections.list(networkInterfaces);
+          Collections.reverse(netIFs);
+          networkInterfaces = Collections.enumeration(netIFs);
+        }
+
         while (networkInterfaces.hasMoreElements()) {
           NetworkInterface ni = networkInterfaces.nextElement();
           Enumeration<InetAddress> addresses = ni.getInetAddresses();
           while (addresses.hasMoreElements()) {
             address = addresses.nextElement();
 
-            // Address must not be link local or loopback. And it must be reachable
+            // Address must not be link wildcard, local or loopback. And it must be reachable
```

Comment:
Good catch! I forgot removing "wildcard" from this comment. Since we have already check the "wildcard" earlier (at Line 206). It's not necessary to filter it in this place. I mean it's not likely getting a "wildcard" address after interface call.